streaming/base/batching/__init__.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Apportion shards/samples to nodes/ranks/workers for elastically deterministic sample order."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from numpy.typing import NDArray
+
+from streaming.base.batching.per_stream import generate_work_per_stream_batching
+from streaming.base.batching.random import generate_work_random_batching
+from streaming.base.world import World
+
+if TYPE_CHECKING:
```

Comment:
Wondering, why do we need this? Were you seeing any exception or was it circular imports ?

Reply:
was getting a circular imports error because I needed `StreamingDataset` for type checking in `generate_work` but then `StreamingDataset` also needed to import `generate_work`. So this takes care of that.